src/status_im/extensions/ethereum.cljs

```diff
@@ -234,12 +234,39 @@
                         :on-failure on-failure}]
      (wrap-with-resolution db json-rpc-args :to execute-send-transaction))))
 
-(defn- parse-log [{:keys [address transactionHash blockHash transactionIndex topics blockNumber logIndex removed data]}]
-  (merge {:data    data
+(defn- event-topic-enc [event params]
+  (let [eventid (str event "(" (string/join "," params) ")")]
+    (abi-spec/sha3 eventid)))
+
+; Return a vector without elements of Y
+(defn diff-vectors [X Y]
+  (reduce #(remove (fn [x] (= x %2)) %1) X Y))
+
+; Return a map with all data params decoded
+(defn- decode-data [data events-hints first-topic indexed]
+  (let [index-event (.indexOf (mapv #(event-topic-enc (% :event) (% :params)) events-hints) first-topic)
+        params-hint (diff-vectors ((get events-hints index-event) :params) indexed) ; Exclude indexed params from decode in data, these are decoded in topics
+        names-hint (mapv keyword ((get events-hints index-event) :names))
+        data-values (mapv (partial apply str) (partition 64 (string/replace-first data #"0x" "")))]
+    (zipmap names-hint (mapv abi-spec/hex-to-value data-values params-hint))))
+
+; This assumes that Topic 0 is filtered and return a map with all topics decoded
+(defn- decode-topics [topics indexed]
+  (zipmap [:topic1 :topic2 :topic3] (mapv abi-spec/hex-to-value topics indexed)))
```

Comment:
Only up to `topic3` ?

Reply:
I read from [Events-Solidity](https://solidity.readthedocs.io/en/latest/abi-spec.html#events) that indexed event params may number up to 3 and Topic 0 is filtered thus I did it only up to topic3. Am I wrong? Could are there more than 4 Topics?

Reply:
My bad, you are right. Maybe worth a comment? :)

Reply:
Done! Thanks!